shared/AppInsightsCore/Tests/Unit/src/ApplicationInsightsCore.Tests.ts

```diff
@@ -1908,6 +1908,338 @@ export class ApplicationInsightsCoreTests extends AITestClass {
             }
         });
 
+        this.testCase({
+            name: "should redact basic auth credentials from URL",
+            test: () => {
+                let config = {} as IConfiguration;
+
+                const url = "https://user:password@example.com";
+                if (config.redactUrls){
```

Comment:
nit: If we are assuming that the "default" is true, then we should be checking that it's true -- otherwise when we default to false the test with the if will never get executed....

So either we don't need the if or we should have an explicit assertion that the value is defaulting to true.